data_migration/migrations/0002_data_firearms_act.py

```
@@ -0,0 +1,53 @@
+# Generated by Django 4.1.2 on 2023-01-13 10:20
+
+from django.db import migrations
+
+FIREARMS_ACT_DATA = [
+    {
+        "act": "Section 1 Firearms",
+        "id": 1,
+    },
+    {
+        "act": "Section 1 Shotguns",
+        "id": 2,
+    },
+    {
+        "act": "Section 2 Shotguns",
+        "id": 3,
+    },
+    {
+        "act": "Section 1 Component Parts",
+        "id": 4,
+    },
+    {
+        "act": "Section 1 Ammunition",
+        "id": 5,
+    },
+    {
+        "act": "Expanding Ammunition 5(1A)(f)",
+        "id": 6,
+    },
+    {
+        "act": "Suppressors",
+        "id": 7,
```

why do you have to specifiy the primary key?

---

Because it referenced by a foreign key in `FirearmsActQuantity`. There are ids in the V1 db (extracted using `ActQuantityParser`) but not the info about the acts they refer to. This is stored in the application xml

---

Yer I wasn't sure if bulk_create allows you to set the primary key of if it was silently ignoring them as the id's you have defined happen to match the order they are being inserted.